torch/linalg/__init__.py

```diff
@@ -1438,49 +1642,57 @@
-and returns a namedtuple (Q, R) of tensors such that :math:`\text{input} = Q R`
-with :math:`Q` being an orthogonal matrix or batch of orthogonal matrices and
-:math:`R` being an upper triangular matrix or batch of upper triangular matrices.
+Computes the QR decomposition of a matrix. Supports batched inputs.
 
-Depending on the value of :attr:`mode` this function returns the reduced or
-complete QR factorization. See below for a list of valid modes.
+For a matrix :math:`A`, this is defined as
 
-.. note::  **Differences with** ``numpy.linalg.qr``:
+.. math::
 
-           * ``mode='raw'`` is not implemented
+    A = QR
 
-           * unlike ``numpy.linalg.qr``, this function always returns a
-             tuple of two tensors. When ``mode='r'``, the `Q` tensor is an
-             empty tensor. This behavior may change in a future PyTorch release.
+where :math:`Q` is orthogonal in the real case and unitary in the complex case, and :math:`R` is upper triangular.
 
-.. note::
-          Backpropagation is not supported for ``mode='r'``. Use ``mode='reduced'`` instead.
+The returned decomposition is a named tuple `(Q, R)`.
+If :attr:`input` is a batch of matrices, then `Q`, `R` are also batched with the same
+batch dimensions.
 
-          Backpropagation is also not supported if the first
-          :math:`\min(input.size(-1), input.size(-2))` columns of any matrix
-          in :attr:`input` are not linearly independent. While no error will
-          be thrown when this occurs the values of the "gradient" produced may
-          be anything. This behavior may change in the future.
+The parameter :attr:`mode` controls the shape of the output. If :attr:`input` has shape
+`(*, m, n)`, denoting `k = min(m, n)`
 
-.. note:: This function uses LAPACK for CPU inputs and MAGMA for CUDA inputs,
-          and may produce different (valid) decompositions on different device types
-          or different platforms.
+- :attr:`mode` `= 'reduced'` (default): Returns `(Q, R)` of shapes `(*, m, k)`, `(*, k, n)` respectively.
+- :attr:`mode` `= 'complete'`: Returns `(Q, R)` of shapes `(*, m, m)`, `(*, m, n)` respectively.
+- :attr:`mode` `= 'r'`: Computes only `R`. Returns `(Q, R)` with `Q` empty and `R` of shape `(*, k, n)`.
 
-Args:
-    input (Tensor): the input tensor of size :math:`(*, m, n)` where `*` is zero or more
-                batch dimensions consisting of matrices of dimension :math:`m \times n`.
-    mode (str, optional): if `k = min(m, n)` then:
+Differences with `numpy.linalg.qr`:
+
+- :attr:`mode` `= 'raw'` is not implemented.
+- Unlike `numpy.linalg.qr`, this function always returns a tuple of two tensors.
+  When :attr:`mode` `= 'r'`, the `Q` tensor is an empty tensor.
+  This behavior may change in a future PyTorch release.
 
-          * ``'reduced'`` : returns `(Q, R)` with dimensions (m, k), (k, n) (default)
+.. note:: The elements in the diagonal of `R` are not necessarily positive.
```

Comment:
Is it in general expected that the diagonal elements are positive?

Reply:
I would say so. This is an easy way to make the QR unique, so it is often assumed in the literature. In other functions, such as `eig`, we have similar types of normalisation to make the decompositions unique when possible.